operatorcourier/validate.py

```diff
@@ -24,52 +32,44 @@ def validate(self, bundle):
         return self._bundle_validation(bundle)
 
     def _bundle_validation(self, bundle):
-        validationDict = dict()
-
         if self.dataKey not in bundle:
             logger.error("Bundle does not contain base data field.")
             #break here because there's nothing else to learn
             return False
 
         bundleData = bundle[self.dataKey]
 
-        validationDict[self.crdKey] = self._type_validation(bundleData, self.crdKey, self._crd_validation, False)  
-        validationDict[self.csvKey] = self._type_validation(bundleData, self.csvKey, self._csv_validation, True)
-        validationDict[self.pkgsKey] = self._type_validation(bundleData, self.pkgsKey, self._pkgs_validation, True)
+        if not self._type_validation(bundleData, self.crdKey, self._crd_validation, False):
```

Comment:
why refactor this? right now it is set up specifically so that we can get the range of all errors, rather than fail as soon as one of these validation tests runs.

Reply:
@kevinrizza oh I didn't know it was intentional. I did it for a reason. Because when I check the fields match in csv spec validation, I would expect it passes all the CRD presence check so that I will directly access the field and compare. If some fields are missing then [this line](https://github.com/operator-framework/operator-courier/pull/41/files#diff-4badd63917dd32ab713b1c89ddfa260cR149) will raise an exception which accesses a non-existing field.

Reply:
@jeremylinlin Just reordering those checks would cause your code to fail suddenly. Let's be more careful and when you are accessing fields that could not exist inside the map always check for them to exist first. We should consider each of the checks in the validator to be independent and not come with any previous knowledge. While it's true that it will reduce performance somewhat, we are not really performance bound by these checks right now.